packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -34,10 +34,7 @@ import type { TransformerContext } from '../../types';
 
 const bGenerateSlottedContent = esTemplateWithYield`
         const shadowSlottedContent = ${/* hasShadowSlottedContent */ is.literal}
-            ? async function* generateSlottedContent(instance) {
-                // The 'instance' variable is shadowed here so that a contextful relationship
-                // is established between components rendered in slotted content & the "parent"
-                // component that contains the <slot>.
+            ? async function* generateShadowSlottedContent() {
```

Comment:
This `instance` shadowing seems wrong to me, and is probably why removing this fixed so many tests. Slotted content is owned by the slott<em>er</em>, not the slott<em>ee</em>. If you do

```html
<!-- foo.html -->
<template>
  <x-bar>
    {yolo}
  </x-bar>
</template>
```

Then `yolo` is owned by `<x-foo>`, not `<x-bar>`. `this.yolo` should be evaluated relative to `<x-foo>`.